### PR TITLE
RFC: from __future__ import annotations + pyupgrade

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

**Review by asottile — APPROVED** [review]
noice

File: doc/en/conf.py
Comment (nicoddemus) [inline doc/en/conf.py:400]: Did pyupgrade did this, or you updated it manually?

**Review by nicoddemus — APPROVED** [review]

File: doc/en/conf.py
Comment (RonnyPfannschmidt) [reply doc/en/conf.py:400]: Pyupgrade removes stringification on type declarations when adding the annotations future

Comment (nicoddemus) [reply doc/en/conf.py:400]: Noice

**Review by bluetech — APPROVED** [review]
Would be nice to be able to use the new syntax!

**Comment by Pierre-Sassoulas** [comment]
> Any idea how we can work with that (or do we backport this one's)

Yeah, it needs either a backport itself or to wait for the release of the next minor being imminent before merging. Depending on the confidence we have it's not going to break the maintenance branch..

**Comment by nicoddemus** [comment]
> Depending on the confidence we have it's not going to break the maintenance branch..

**Comment by RonnyPfannschmidt** [comment]
Let's give it a call out, we might Learn of really interesting (or horrifying) use cases for the annotations

Im going to take a look at fixup /backport fixup soonish

**Comment by RonnyPfannschmidt** [comment]
@nicoddemus @bluetech now that 8x. and 7.4 had some major releases, i'd like to propose porting main and 8.x over to the new syntax, i'll try to get it tried this weekend

**Comment by RonnyPfannschmidt** [comment]
this pr as is may be a complete pain to review - i wonder if i ought to split it across some logical/file boundaries

File: src/_pytest/_code/code.py
Comment (bluetech) [inline src/_pytest/_code/code.py:718]: Would wrap in parens to make the formatting less weird

File: src/_pytest/_code/code.py
Comment (bluetech) [inline src/_pytest/_code/code.py:748]: Would wrap in parens to make the formatting less weird

File: src/_pytest/fixtures.py
Comment (bluetech) [inline src/_pytest/fixtures.py:980]: Redundant parens

File: src/_pytest/fixtures.py
Comment (bluetech) [inline src/_pytest/fixtures.py:1189]: Redundant parens

File: src/_pytest/fixtures.py
Comment (bluetech) [inline src/_pytest/fixtures.py:1233]: Redundant parens

File: src/_pytest/fixtures.py
Comment (bluetech) [inline src/_pytest/fixtures.py:1246]: Redundant parens

File: src/_pytest/fixtures.py
Comment (bluetech) [inline src/_pytest/fixtures.py:1258]: Redundant parens

File: src/_pytest/mark/structures.py
Comment (bluetech) [inline src/_pytest/mark/structures.py:475]: Redundant parens & weird formatting

File: src/_pytest/python.py
Comment (bluetech) [inline src/_pytest/python.py:1166]: Redundant parens

**Review by bluetech — APPROVED** [review]
A few formatting comments but LGTM. I suggest merging quickly to avoid conflicts.

**Comment by nicoddemus** [comment]
> I suggest merging quickly to avoid conflicts.

Agree, I won't have time to review this in a timely manner, but please go ahead. Also I suggest to squash the commits into a single one, as the last two are just fixups.

**Comment by RonnyPfannschmidt** [comment]
closing in favor of #12467

**Pull request closed without merging**
